frontend-app/src/ToDoList.js

```diff
@@ -0,0 +1,85 @@
+import { useState } from "react";
+
+function ToDoList() {
+    const [input, setInput] = useState('');
+    const [todos, setToDo] = useState([]);
+
+    const onInputChange = (e) => {
+        setInput(e.target.value);
+    }
+
+    const addToDo = (e) => {
+        e.preventDefault();
+        if (input === '') {
+            return;
+        }
+
+        const newToDo = [...todos, { value: input, isDone: false }];
+        setToDo(newToDo);
+        setInput('');
+    }
+
+    const deleteToDo = index => {
+        const newToDos = [...todos];
+        newToDos.splice(index, 1);
+        setToDo(newToDos);
+    };
+
+    const deleteToDoAll = () => {
+        setToDo([]);
+    }
+
+
+    const toggleComplete = (index) => {
+        const newToDo = [...todos];
+        newToDo[index].isDone = !newToDo[index].isDone;
+        setToDo(newToDo);
+    }
+
+    return (
+        <div className="container">
+            <h1 className="text-center">To Do List</h1>
+            <div className="justify-content-center">
+                <form className="input-group" onSubmit={addToDo} >
+                    <input onChange={onInputChange} value={input} type="text" className="form-control" />
+                    <div className="input-group-append">
+                        <button className="input-group-text">Add</button>
+                    </div>
+                </form>
+                <div >
+                    <ul className="list-group">
+
+                        {todos.map((todo, index) =>
+                            <ToDo
+                                key={index}
+                                todo={todo}
+                                toggle={() => toggleComplete(index)}
+                                value={todo.value}
+                                isDone={todo.isDone}
+                                del={() => deleteToDo(index)}
+                            />)
+                        }
+                    </ul>
+                </div>
+            </div>
+            <button className="input-group-text" onClick={deleteToDoAll}>Delete all</button>
+        </div>
+    );
+}
+
+function ToDo({ value, isDone, toggle, del, index }) {
```

Comment:
индекс уже же не нужен
```suggestion
function ToDo({ value, isDone, toggle, del }) {
```